app/src/marketplace/components/Modal/PurchaseSummaryDialog/index.jsx

```diff
@@ -1,64 +1,47 @@
 // @flow
 
-import React from 'react'
+import React, { useMemo } from 'react'
 import { Translate, I18n } from 'react-redux-i18n'
 import BN from 'bignumber.js'
 
 import ModalPortal from '$shared/components/ModalPortal'
 import Dialog from '$shared/components/Dialog'
-import type { TimeUnit, PaymentCurrency, NumberString } from '$shared/flowtype/common-types'
-import { paymentCurrencies, contractCurrencies } from '$shared/utils/constants'
-import { formatDecimals, dataToUsd } from '$mp/utils/price'
-import { isMobile } from '$shared/utils/platform'
+import type { TimeUnit, PaymentCurrency } from '$shared/flowtype/common-types'
+import { contractCurrencies } from '$shared/utils/constants'
+import { formatDecimals } from '$mp/utils/price'
 
 import styles from './purchaseSummaryDialog.pcss'
 
 export type Props = {
     name: string,
     time: string,
     timeUnit: TimeUnit,
+    paymentCurrency: PaymentCurrency,
     price: BN,
-    ethPrice: BN,
-    daiPrice: BN,
-    dataPerUsd: NumberString,
-    ethPriceInUsd: NumberString,
-    purchaseStarted: boolean,
+    approxUsd: BN,
+    purchaseStarted?: boolean,
+    waiting?: boolean,
+    onBack: () => void,
```

Comment:
Added `onBack` to go back to the previous screen, curiously this is only available in the mobile version of design.

Reply:
Hmm true, @mattatgit was this intentional? 

**_Complete your purchase_** has a back button on mobile but not on desktop - 
https://share.goabstract.com/ce78fb23-6e4e-472b-9c8f-f677d490b757?mode=design&sha=22fda9a67b422eb66e3a86d23687ca30a90593e5

https://share.goabstract.com/06545072-fcab-4bc0-9e93-1a85e5bc3705?mode=design&sha=22fda9a67b422eb66e3a86d23687ca30a90593e5

Reply:
@fonty1 Looks like just a miss. User should be able to go back in case they set the wrong period of access. If the desktop "Cancel" button/link is replaced (on just that view) with Back, should be fine

Reply:
@fonty1 Quick note, if you can include screenshots (of the designs) for these kind of questions i can usually just look at GH & reply, making the process faster. 

If links to Abstract, i often have to sign in to the web app (because the desktop app doesn't intercept links), which can be either slow (if on laptop) or not possible if on phone.

Not a biggie, just an FYI.

Reply:
Can do for next time 👍 Thanks for the late night check!

Reply:
Fixed in https://github.com/streamr-dev/streamr-platform/pull/1012/commits/1c22a39c4870fc945440a6c83592d82f605ffa22